src/targets/gpu/lowering.cpp

```diff
@@ -236,11 +238,39 @@ struct miopen_apply
             {
                 check_shape(s, insert_precompile_op(it));
             }
+            else if(it->get_operator().has_offload_copy()) // custom op
```

Comment:
This should just be an `else`.

Reply:
`else if` does assertion that it is indeed custom op.

Reply:
I can add `else` clause that throws, if it is better.

Reply:
This logic should be applied to all cpu only operators not just custom ops.

Reply:
cpu operators you mean e..g `NMS` ? it is part of `apply_map` and handled in first `if` condition.

Reply:
> This logic should be applied to all cpu only operators not just custom ops.

CPU operators require allocation while for the custom_op we are assuming they are provided.    
Also, can't keep just the else clause, this  `apply()` function is traversing through module, therefore contains @literals, @params etc. also.

Reply:
> CPU operators require allocation while for the custom_op we are assuming they are provided.

I dont really follow this.

> Also, can't keep just the else clause, this apply() function is traversing through module, therefore contains @literals, @params etc. also.

Yea, just check for the `target` attribute.